vllm/v1/worker/metal_phaseprof.py

```
@@ -55,6 +62,159 @@ def phase(name: str):
+    finally:
+        _pyprof.disable()
+        _pyprof_steps += 1
+        if _pyprof_steps % _DUMP_EVERY == 0:
+            _dump_pyprof()
+
+
+def _dump_pyprof() -> None:
+    import io
+    import pstats
+
+    path = f"/tmp/pyprof_{os.getpid()}.txt"
```

Ruff (0.16.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/pyprof_"

(S108)